src/main/java/org/opensearch/sdk/handlers/ExtensionsInitRequestHandler.java

```diff
@@ -49,9 +54,21 @@ public InitializeExtensionResponse handleExtensionInitRequest(InitializeExtensio
         logger.info("Registering Extension Request received from OpenSearch");
         extensionsRunner.opensearchNode = extensionInitRequest.getSourceNode();
         extensionsRunner.setUniqueId(extensionInitRequest.getExtension().getId());
+
```

Comment:
per previous comment on the test case, I think all this new code would fit nicely in a method on the ExtensionsRunner

Reply:
Yes, changes are pushed